pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -63,9 +64,10 @@ func MakeIngressVirtualService(ia *v1alpha1.Ingress, gateways map[v1alpha1.Ingre
 		vs.Labels = make(map[string]string)
 	}
 
-	ingressLabels := ia.GetLabels()
-	vs.Labels[serving.RouteLabelKey] = ingressLabels[serving.RouteLabelKey]
-	vs.Labels[serving.RouteNamespaceLabelKey] = ingressLabels[serving.RouteNamespaceLabelKey]
+	vs.Labels = resources.FilterMap(ia.GetLabels(), func(k string) bool {
+		return k != serving.RouteLabelKey && k != serving.RouteNamespaceLabelKey
+	})
+	vs.Labels[networking.IngressLabelKey] = ia.Name
```

Comment:
I think that previously this just copied these two down, but now it creates a fully copy of the ingress labels, is that right?  Perhaps we should just set the IngressLabelKey to reduce churn?

The code above can be cleaned up to simply inline:
```
Labels: map[string]string{
  networking.IngressLabelKey: ia.Name,
},
```

... inline in the ObjectMeta.

Reply:
I see, maybe I read the filter backwards?

Reply:
Yes, the predicate is for removal.